From c3db937348e445988453b40e7e33dd6e3516e998 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C3=A5rten=20Kongstad?= <marten.kongstad@sonymobile.com>
Date: Thu, 2 Jun 2016 09:35:09 +0200
Subject: [PATCH 02/31] OMS-N: installd: add command 'rmidmap' [2/14]

Add an installd command to remove an idmap file. This is the inverse of
the 'idmap' command and is intended for clean-up once an idmap file is
no longer needed because an APK was removed, etc.

This commit depends on a corresponding commit in frameworks/native
(with the same Change-Id).

Bug: 31052947
AOSP Change-Id: Iae19a519803f0c172b02a32faa283ef36f43863c

Change-Id: Ia91a0d63b6a8e12173af671a83482969aa5f4d03
---
 services/core/java/com/android/server/pm/Installer.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/services/core/java/com/android/server/pm/Installer.java b/services/core/java/com/android/server/pm/Installer.java
index 9cea3b2..5198a2a 100644
--- a/services/core/java/com/android/server/pm/Installer.java
+++ b/services/core/java/com/android/server/pm/Installer.java
@@ -168,6 +168,13 @@ public final class Installer extends SystemService {
         mInstaller.execute("idmap", targetApkPath, overlayApkPath, uid);
     }
 
+    public void removeIdmap(String overlayApkPath) throws InstallerException {
+        StringBuilder builder = new StringBuilder("rmidmap");
+        builder.append(' ');
+        builder.append(overlayApkPath);
+        mInstaller.execute(builder.toString());
+    }
+
     public void rmdex(String codePath, String instructionSet) throws InstallerException {
         assertValidInstructionSet(instructionSet);
         mInstaller.execute("rmdex", codePath, instructionSet);
-- 
2.8.2

